### ISSUE TITLE
Client is still not ready after resolving "client.startClient()" promise 

### ISSUE THREAD

**Issue by AndreasGassmann** [matrix-org/matrix-js-sdk#1218]
I'm trying to establish a connection and once that is done, I would like to immediately send a message.

If I just have this code then it doesn't work (I don't remember the exact error, if this is a bug I can provide the error message as well):

```
await client.startClient({ initialSyncLimit: 0 })

// send message
```

What I have to do to make it work is this:

```
await client.startClient({ initialSyncLimit: 0 })

await new Promise(resolve => setTimeout(resolve, 1000))

// send message
```

Is it intended that the client is not fully ready after the `client.startClient()` promise resolves? Do I have to wait for another event or is this a bug?

**Comment by t3chguy** [comment]
> If I just have this code then it doesn't work (I don't remember the exact error, if this is a bug I can provide the error message as well):

The error message is important. You should be able to send a message before `startClient` as startClient just starts syncing but sending should be unaffected.

**Comment by t3chguy** [comment]
> Is it intended that the client is not fully ready after the client.startClient() promise resolves? Do I have to wait for another event or is this a bug?

Yes, not fully ready as you are yet to await sync so your local stored data may still be empty, but this would not matter for sending.

**Comment by AndreasGassmann** [comment]
Thanks for the quick reply. I will post the error message asap.
